README.md

```diff
@@ -1,53 +1,109 @@
-the token to your repository secrets so that it can be used in your workflow.
+☁️ If you're pushing to a wiki that's **not the current repository** you'll need
+to get a [GitHub PAT] to push to it.
 
-Simple example usage can be found in [this workflow file](./.github/workflows/update-wiki.yml).
+### Options
+
+<!--
+Prettier doesn't format inline HTML. See #8480 on GitHub Issues. [1] Instead,
+You'll need to either wrap this in an ```html code block to abuse the formatter
+and then remove the ```html wrapper, or paste it into the online Prettier
+playground [2] and copy the result.
+
+[1]: https://github.com/prettier/prettier/issues/8480
+[2]: https://prettier.io/playground/
+-->
+
+<dl>
+  <dt><code>repository</code></dt>
+  <dd>
+    <b>Default:</b> <code>github.com/${{ github.repository }}</code><br />
+    The repository housing the wiki
+  </dd>
+  <dt><code>token</code></dt>
+  <dd>
+    <b>Default:</b> <code>${{ github.token }}</code><br />
+    The token to use when cloning and pushing wiki changes
+  </dd>
+  <dt><code>wiki-directory</code></dt>
+  <dd>
+    <b>Default:</b> <code>.github/wiki</code><br />
+    The directory to use for your wiki contents
+  </dd>
+  <dt><code>commit-message</code></dt>
+  <dd>
+    <b>Default:</b> <code>Update wiki ${{ github.sha }}</code><br />
+    The message to use when committing to the wiki
+  </dd>
+</dl>
 
 ## Alternatives
 
-There are numerous GitHub wiki publishing actions on the GitHub Actions
-marketplace. There are, however, two that stick out. The
-[newrelic/wiki-sync-action][newrelic-wiki-sync-action] is a good choice for a
-GitHub wiki action if you need bidirectional synchronization when someone edits
-the live wiki. This can be beneficial for non-technical contributors. There's also
-[Andrew-Chen-Wang/github-wiki-action][andrew-chen-wang-github-wiki-action]
-which is a direct competitor to this project. The tradeoff that it makes is using a
-Docker image which may not work with Windows-based jobs.
-
-For more information about other alternatives, you can check out [Issue #4][issue-4]
-which has a longer list of alternatives.
-
-[newrelic-wiki-sync-action]: https://github.com/newrelic/wiki-sync-action#readme
-[andrew-chen-wang-github-wiki-action]: https://github.com/Andrew-Chen-Wang/github-wiki-action#readme
-[issue-4]: https://github.com/spenserblack/actions-wiki/issues/4
+There are quite a few GitHub wiki publishing actions on the [GitHub Actions
+marketplace]. There are, however, two that stick out. The
+[newrelic/wiki-sync-action] is a good choice for if you need bidirectional
+synchronization when someone edits the live wiki. This can be beneficial for
+less-technical contributors. There's also [Andrew-Chen-Wang/github-wiki-action]
+which is a direct competitor to this project. Specifically, it offers more
+automagic ✨ features in exchange for more complexity.
+
+📚 If you're interested in more discussion of alternatives, check out [#4].
+
+<!-- prettier-ignore-start -->
```

Comment:
Why not just let prettier format these, and then use
```markdown
[foo/bar][foo-bar]
```
?

Reply:
The reason I use `[term] then text` instead of `[term][id-desc-here] then text` is because that ID can be long and has the same problems as `[term](https://example.org/long-link) then text`: it causes wrapping where wrapping looks bad and is un-ergonomic to editing. It comes from experience writing documents with a lot of linking (think specifications where you link to each term as a definitive "this is a reference to a specific thing defined elsewhere") and may not be applicable here, but I find that the style transfers well to make a pleasant editing experience.

```
Hello [world], this is a brand new idea! I have made
[a new thing] that I want you to try out! It can do
[many things] such as [do a loop-de-loop] and [perform
tricks]! It even [works outside]! I got inspiration
from [George Washington]'s [address at the Battle of
Gettysburg] where he discussed the [Google] stock price.

<!-- lots of links here -->
[...]: ...
```

```md
Hello [world][earth], this is a brand new idea! I have made [a new
thing][the-new-thing] that I want you to try out! It can do [many
things][feature-list] such as [do a loop-de-loop][feature-loop-de-loop] and
[perform tricks][feature-perform-tricks]! It even [works
outside][feature-works-outside]! I got inspiration from [George
Washington][george-washington]'s [address at the Battle of
Gettysburg][gettysburg-address] where he discussed the [Google][google] stock
price.

<!-- lots of links here -->
[...]: ...
```

Note how the `endingword][moderate-length-id]` is all treated as _one unwrap-able token_? That's the part that I dislike since it makes the lines uneven and sometimes only a few words. This means that _skimming is harder_ since that break might be an actual two line paragraph break, but might also be a long wrapping link. This is also why I dislike inline `[text](https://)` links.

If you're asking why I use `<!-- prettier-ignore-start -->` on the `[link]: https://` part, that's because prettier does this:

```md
[hello world]:
  https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#about-yaml-syntax-for-workflows
[thing]:
  https://learn.microsoft.com/en-us/cpp/standard-library/iostream?view=msvc-170#global-stream-objects
[this is one line]: https://docs.github.com/en
[long link text short link but still]:
  http://web.ihep.su/dbserv/compas/src/bohr13/eng.pdf
```

and I think this is more readable, hence the ignore:

```md
[hello world]: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#about-yaml-syntax-for-workflows
[thing]: https://learn.microsoft.com/en-us/cpp/standard-library/iostream?view=msvc-170#global-stream-objects
[this is one line]: https://docs.github.com/en
[long link text short link but still]: http://web.ihep.su/dbserv/compas/src/bohr13/eng.pdf
```

Reply:
I use a vertical monitor and disable soft wrapping, so overly long lines are absolutely a pain point that I hate :laughing:
Several times I've put a link on its own line if it was too long.

But, yeah, personally, I'd like to see something like this:
```markdown
[#4][issue-4]

[issue-4]: https://...
```

As long as the IDs aren't overly long, this shouldn't be a problem. And feel free to add a hard wrap on long lines! I don't mind seeing something like this:
```markdown
This is a sentence
[with a veeeeeeeeeeeeeeeeeeeeeery loooooooooooong liiiiiiiiiiiiiiink!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!1][link-1]
```

Reply:
I use VS Code and have Prettier activated all the time, which means that when I save (I also have format-on-save enabled) it transforms your nice example into this:

```md
This is a sentence [with a veeeeeeeeeeeeeeeeeeeeeery loooooooooooong
liiiiiiiiiiiiiiink!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!1][link-1]
```

which, you're correct, works like normal for short ids. But even for short ids, if the last token is oddly sized, it creates things like this:

```md
I hope that you have enjoyed this! Our [HTML
Specification-v2.0][whatwg-html-v2.0] is not yet ready for the prime time, so
try to be patient with us as we make sure that it meets all the necessary
```

its the sort of issue that i find extra annoying since it only happens just often enough to be irritating, but also rare enough that it doesn't come up in every writing session.

my 2 cents. if you want id-style links, I guess OK then will do 👍

Reply:
Another thing is I find it needlessly repetitive. Like the terms are case-insensitive, they don't need to be defined right away, and they are shorter, so why would I bother _going back o a 2nd or 3rd pass_ just to add in pretty link ids when the literal `[text in here]` is already a good id?? 😆

Reply:
Another point (sorry for the literal dog pile here, I just keep thinking of pro/cons) is that when you use GitHub Copilot, and you get to the bottom of your document after typing for like 3 minutes with a bunch of un-defined `[terms] like [this]` that don't have `[terms]: https://`, you: 1) get GitHub Copilot autocompletion so you don't have to hunt, it just generates the `[terms]: ` prefix, you put in the url so that you 2) don't even have to go back and remember/copy+paste what those potentially multi-word ids were like `[term that includes many words]` just gets autocompleted! 😱

Reply:
We really should take this discussion somewhere else ✈ since we are spiraling into workflow stuff and completely derailed from topic of code review 😁

Reply:
A bit off-topic for this PR, but would you be willing to make a PR to set up `.devcontainer/devcontainer.json`? If we, and other contributors, have a common setup for our editors, it might be easier to agree on certain styles and dev experiences. I'd be happy to offer advice if you have questions about setting up codespaces.